### DISCUSSION TITLE
Recovering from a split mesh calc

### DISCUSSION THREAD

**Discussion by makeclean** [idaholab/moose#19429] [Q&A General] [answered]
I've got a problem that I split into 36 bits using `split-mesh`, I turned recovery on, since I knew this calc was going to run longer than my HPC's queue time. It split fine, and ran fine with the following command:
```
mpirun -np 36 --npernode 6 <exec> -i <input_file> --use-split
```
I included a recovery option in the input,   
```
[Outputs]
  [checkpoints]
      type = Checkpoint
      num_files = 2
      interval = 40
  []
[]
```
However, when trying to restart I get consisent failures; using the command
```
mpirun -np 36 --npernode 6 <exec> -i <input_file> --use-split --recover
```
I get the error, 
```
** ERROR ***
ERROR: Neither one of the following files can be located:
	'_mesh.cpr/36/header.cpr' nor
	'_mesh.cpr'
If you are running a parallel job, double check that you've created a split for 36 ranks.
Note: One of paths above may refer to a valid directory on your system, however we are attempting to read a valid header file.
```
Even setting the split mesh filename `--split-file tpipe.cpr` makes no difference to the error. In the checkpoint directory for the filename, which happens to be `tpipe-cfl-water-temp-correct-pc-6-36_checkpoints_cp/1920_mesh.cpr`. 

Is there a better way/right way to restart properly?

**Comment by GiudGiud** [comment, marked as answer]
I think that s the way but maybe someone will correct me

Can you get this to run by renaming the folder to the name it s expecting? _mesh.cpr it seems

**Reply by makeclean** [reply]
:sad face: afraid not, 
```
*** Info ***
Using  for recovery.
Setting Up
  Setting Mesh...                                                                        [
 27.30 s] [ 1270 MB]
Finished Setting Up                                                                      [
 29.15 s] [ 1339 MB]

*** ERROR ***
Unable to open file ".rd-47". Check to make sure that it exists and that you have read per
mission.
```
I made a symbolic link from the timestep number/num splits folder to `_mesh.cpr` as suggested, its a hard no.

**Reply by GiudGiud** [reply]
ah it is looking for an empty filename now. There might be broken logic. I ll see if I can repoduce this

**Reply by GiudGiud** [reply]
Ok so I think the problem is that you did not specify the file name after --recover
I m going to think about this more but maybe we want to error here when the file name is missing. Not sure why " " was an OK default

**Reply by GiudGiud** [reply]
Note that once the mesh is split, I think the recover use that directly, no need to add --use-split --split-file split-mesh-filename anymore, since
--recover input_name_checkpoint_cp/00014 (which is akin to what you need to specify here) is already a distributed file. 

Splits are sort of a confusing additional distinction in general. It's really distributed vs replicated mesh that is the big deal. Using a split or not only makes a difference at initialization, on slow file systems. I think this question is more about a distributed recover than a split recover.
